File: src/SharpOMatic.Engine/DTO/EvalRunRowDetail.cs
namespace SharpOMatic.Engine.DTO;

public class EvalRunRowDetail
{
    public required Guid EvalRunRowId { get; set; }
    public required Guid EvalRunId { get; set; }
    public required Guid EvalRowId { get; set; }
    public required string Name { get; set; }
    public required int Order { get; set; }
    public required EvalRunStatus Status { get; set; }
    public required DateTime Started { get; set; }
    public DateTime? Finished { get; set; }
    public string? OutputContext { get; set; }
    public string? Error { get; set; }
    public required List<EvalRunRowGraderDetail> Graders { get; set; }
}

File: src/SharpOMatic.FrontEnd/src/app/eval/definitions/eval-run-row-detail.ts
import { EvalRunStatus } from '../enumerations/eval-run-status';

export interface EvalRunRowGraderDetailSnapshot {
  evalRunRowGraderId: string;
  evalRunRowId: string;
  evalGraderId: string;
  label: string;
  order: number;
  status: EvalRunStatus;
  started: string;
  finished: string | null;
  score: number | null;
  payload: string | null;
  error: string | null;
}

export interface EvalRunRowDetailSnapshot {
  evalRunRowId: string;
  evalRunId: string;
  evalRowId: string;
  name: string;
  order: number;
  status: EvalRunStatus;
  started: string;
  finished: string | null;
  outputContext: string | null;
  error: string | null;
  graders: EvalRunRowGraderDetailSnapshot[];
}
